### PR TITLE
no bullet shape for collisionless polyvox

### PR DESCRIPTION
**Pull request by sethalves**
- collisionless PolyVox will no longer generate a shape for bullet

### CODE REVIEW

**Comment by sethalves** [comment]
to test:
- make sure voxels that are not collisionless still collide.  If you don't know of any, try these:
[voxel terrain](hifi://eschatology/79.6513,-0.999147,180.205/0,0.0403851,0,0.999184)
- find a voxel-paint setup and make sure it works.  perhaps here:
[voxel-paint](hifi://eschatology/80.7164,33.4602,104.756/0,0.999269,0,-0.0382219)

**Comment by hifi-gustavo** [comment]
Build console output for commit: 422b0bb952df78b51a59166e66ef3ef2f52d4a8e available at: https://gist.github.com/ff8c6a5bceac9ef7178ee709989ff8e4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9759-422b0bb.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9759-422b0bb.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1db7d8b116cdb6d86d31cae2367fe418a8f4ade3 available at: https://gist.github.com/ff8c6a5bceac9ef7178ee709989ff8e4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9759-1db7d8b.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9759-1db7d8b.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6227db9fc0032c2cf6e29610bd46f6d562daf99a available at: https://gist.github.com/ff8c6a5bceac9ef7178ee709989ff8e4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9759-6227db9.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9759-6227db9.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 542cb7ab85aa61558fa50b0feb130f56a11b4842 available at: https://gist.github.com/ff8c6a5bceac9ef7178ee709989ff8e4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9759-542cb7a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9759-542cb7a.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 88c850afa233882d2cb68effd43822502707e755 available at: https://gist.github.com/ff8c6a5bceac9ef7178ee709989ff8e4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9759-88c850a.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9759-88c850a.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 1f402f7e4e783fffbbc91e27fe3f9e41bdd9a28e available at: https://gist.github.com/ff8c6a5bceac9ef7178ee709989ff8e4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9759-1f402f7.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9759-1f402f7.dmg

File: libraries/shared/src/GeometryUtil.cpp [hunk 1]
```diff
@@ -205,6 +205,33 @@ bool findRaySphereIntersection(const glm::vec3& origin, const glm::vec3& directi
...
+        if (originProjection < 0.0f) { // below start
+            return pointInSphere(origin, start, radius);
+        } else if (originProjection > capsuleLength) { // above end
+            return pointInSphere(origin, end, radius);
+        } else { // between start and end
+            return true;
+        }
+    }
+    return false;
+}
```
Comment (AndrewMeadows) [inline libraries/shared/src/GeometryUtil.cpp]: These kinds of tests should be optimized if possible.  I'll try to write up some faster implementations today and submit as PR against this one.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6b230bad1453b025c2438cebcea875580187f8d0 available at: https://gist.github.com/ff8c6a5bceac9ef7178ee709989ff8e4<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9759-6b230ba.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9759-6b230ba.dmg

**Comment by nbq** [comment]
testing

**Comment by nbq** [comment]
Tester: Coal
System: i7 6700k, GeForce 970, Windows 10

* Went to `hifi://eschatology/79.6513,-0.999147,180.205/0,0.0403851,0,0.999184` and verified that I could walk on the polyvox terrain as I should be able to
* Went to `hifi://eschatology/80.7164,33.4602,104.756/0,0.999269,0,-0.0382219` and tried the voxel paint script and verified I could still paint with it
* Even set something I painted to collisionless to verify I could not walk through it and I could not, before I could.

**SUCCESS PASS**

**Pull request merged**